src/util/enumerate.rs

```diff
@@ -0,0 +1,80 @@
+use futures::{Async, Poll, Stream, Sink, StartSend};
+
+/// A stream combinator which combines the yields the current item
+/// plus its count starting from 0.
+///
+/// This structure is produced by the `Stream::enumerate` method.
+#[derive(Debug)]
+#[must_use = "streams do nothing unless polled"]
+pub struct Enumerate<T> {
+    stream: T,
```

Comment:
since there is a sink impl I think it makes sense to name this `inner` same with the must use comment.